lang/funcs/core/example/str2float_func.go

```diff
@@ -0,0 +1,40 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package coreexample
+
+import (
+	"strconv"
+
+	"github.com/purpleidea/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	simple.ModuleRegister(moduleName, "str2float", &types.FuncValue{
+		T: types.NewType("func(a str) float"),
+		V: func(input []types.Value) (types.Value, error) {
+			var i float64
+			if val, err := strconv.ParseFloat(input[0].Str(), 64); err == nil {
```

Comment:
reverse to `!=` please and i think the code looks much cleaner.

Reply:
Also please move function to below with a docstring so that it will appear in docs.

Reply:
Example: https://github.com/purpleidea/mgmt/blob/38dfaa1caaafb527dc09ed2a0736e4d2b25cf5e4/lang/funcs/core/math/pow_func.go#L35